### PR TITLE
Refactor Card component test file to new format

### PR DESCRIPTION
**Pull request by adi-unni**
### What is the context of this PR?

Fixes: [#141](https://github.com/ONSdigital/design-team/issues/141)

Refactors the card component test file to the new agreed format.

### How to review this PR

Run tests locally and check that the messages make sense and describe what is being tested

Ask yourself....

- Do we need any additional tests?
- Are the tests descriptions clear?
- Does the order of the test document make sense?
- Is it internally consistent?
- Is it consistent with the [guidelines for test refactoring (https://confluence.ons.gov.uk/display/DS/Testing+Refactor+Guidance)?
- run this test command to view the test results:
- yarn test --testNamePattern="FOR: Macro: Card"

-   [x] I have selected the correct Assignee
-   [x] I have linked the correct Issue

### CODE REVIEW

File: src/components/card/_macro.spec.js [hunk 1]
```diff
@@ -1,268 +1,246 @@
...
-                },
+            test('THEN: the provided `text` is accessible via the `textId` identifier', () => {
+                expect($('#example-text-id').text().trim()).toBe('Example card text.');
             });
+        });
 
-            expect(listsSpy.occurrences[0]).toEqual({
-                variants: 'dashed',
-                itemsList: [{ text: 'Test item 1' }, { text: 'Test item 2' }],
+        describe('WHEN: heading levels are tested', () => {
```
Comment (rmccar) [inline src/components/card/_macro.spec.js:37]:
```suggestion
        describe('WHEN: headingLevel is provided', () => {
```

File: src/components/card/_macro.spec.js [hunk 2]
```diff
@@ -1,268 +1,246 @@
...
+            it.each([
+                [1, 'h1'],
+                [4, 'h4'],
+            ])('THEN: it has the correct element headingLevel tag %i outputs <%s> tags', (headingLevel, expectedTag) => {
```
Comment (rmccar) [inline src/components/card/_macro.spec.js:41]:
```suggestion
            ])('THEN: it renders with the headings with the correct heading tag (%i -> <%s>)', (headingLevel, expectedTag) => {
```

**Review by rmccar — CHANGES_REQUESTED** [review]
Theres a few pointers here. Ive gone through half of the changes and created suggestions. I think there needs to be some more refactoring here to get it in the right format. Have a look at some of the other tests for structure and wording for these tests. Access code is a good example

**Review by janehughes — APPROVED** [review]
Looks good, working as expected 👍  I was wondering if it's worth adding a test to check for duplicate IDs if we have multiple cards on the same page?

**Comment by rmccar** [comment]
> Looks good, working as expected 👍 I was wondering if it's worth adding a test to check for duplicate IDs if we have multiple cards on the same page?

Its a good point but we don't have the functionality build into the component to deal with that at the moment. Also I feel this would be above the component level to check and if we were to look into that we would need to implement it on many components. Maybe something to explore at some point but at the moment we trust the service to not duplicate ids

**Review by admilne — APPROVED** [review]
